hts.c

```diff
@@ -938,10 +939,19 @@ htsFile *hts_open_format(const char *fn, const char *mode, const htsFormat *fmt)
          fmt->format == fastq_format))
         fp->format.format = fmt->format;
 
-    if (fmt && fmt->specific)
-        if (hts_opt_apply(fp, fmt->specific) != 0)
+    if (fmt && fmt->specific) {
+        int save_errno = errno;
+        if (hts_opt_apply(fp, fmt->specific) != 0) {
+            if (((hts_opt*)fmt->specific)->opt == CRAM_OPT_REFERENCE) {
+                const char *ref = ((hts_opt*)fmt->specific)->val.s;
+                hts_log_error("Couldn't load reference \"%s\" : %s",
```

Comment:
Having tested this, it looks like `refs_load_fai` already reports this error, so we probably don't need to do it again here after all.  Although making `refs_load_fai` print the error message might be useful:

```
samtools stats --reference /foo 9305_4#1c.cram
[E::fai_build3_core] Failed to open the file /foo
[E::refs_load_fai] Failed to open reference file '/foo'
[E::hts_open_format] Couldn't load reference "/foo" : No such file or directory
[E::hts_open_format] Failed to open file "9305_4#1c.cram"
samtools stats: failed to open "9305_4#1c.cram"
```